src/pool.rs

```diff
@@ -79,16 +81,33 @@ impl serde::Serialize for BackendStats {
     }
 }
 
+/// The name of the pool
+#[derive(Clone, Debug)]
+pub(crate) struct Name(Arc<String>);
```

Comment:
nitpicky: since this just gets `as_str()`'d a lot, why not just represent it as
```suggestion
pub(crate) struct Name(Arc<str>);
```
IIRC this removes a layer of indirection (since the `String` is also a pointer, but `Arc<str>` means the arc points directly to the string), and there's an `From<String> for Arc<str>`  impl that you can use to construct it